data/binding/gen.go

```diff
@@ -380,16 +426,57 @@ func (l *bound{{ .Name }}List) SetValue(i int, v {{ .Type }}) error {
 	if i < 0 || i >= l.Length() {
 		return errOutOfBounds
 	}
-	if l.val != nil {
-		(*l.val)[i] = v
-	}
+
+	l.lock.Lock()
+	(*l.val)[i] = v
+	l.lock.Unlock()
 
 	item, err := l.GetItem(i)
 	if err != nil {
 		return err
 	}
 	return item.({{ .Name }}).Set(v)
 }
+
+func bind{{ .Name }}ListItem(v *[]{{ .Type }}, i int) {{ .Name }} {
+	return &bound{{ .Name }}ListItem{val: v, index: i, old: (*v)[i]}
+}
+
+type bound{{ .Name }}ListItem struct {
+	base
+
+	val   *[]{{ .Type }}
+	index int
+	old   {{ .Type }}
+}
+
+func (b *bound{{ .Name }}ListItem) Get() ({{ .Type }}, error) {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+
+	return (*b.val)[b.index], nil
+}
+
+func (b *bound{{ .Name }}ListItem) Set(val {{ .Type }}) error {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+
+	(*b.val)[b.index] = val
+
+	b.trigger()
+	return nil
+}
+
+func (b *bound{{ .Name }}ListItem) setIfChanged(val {{ .Type }}) error {
```

Comment:
When this is called by list it is holding the listbinding mutex, but it is not holding the base binding mutex. There is a race condition between `item, err := listBinding.GetItem(i); item.Set(value)` and `listBinding.Set(i, value)`

Reply:
Thanks, fixed